OpenRA.Mods.Common/Traits/Render/WithTurretAimAnimation.cs

```diff
@@ -0,0 +1,76 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2017 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Linq;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits.Render
+{
+	public class WithTurretAimAnimationInfo : ConditionalTraitInfo, Requires<WithSpriteTurretInfo>, Requires<ArmamentInfo>, Requires<AttackBaseInfo>
+	{
+		[Desc("Armament name")]
+		public readonly string Armament = "primary";
+
+		[Desc("Turret name")]
+		public readonly string Turret = "primary";
+
+		[Desc("Displayed while targeting.")]
+		[SequenceReference] public readonly string Sequence = null;
+
+		[Desc("Shown while reloading.")]
+		[SequenceReference(null, true)] public readonly string ReloadPrefix = null;
+
+		public override object Create(ActorInitializer init) { return new WithTurretAimAnimation(init, this); }
+
+		public override void RulesetLoaded(Ruleset rules, ActorInfo ai)
+		{
+			var turretAttackAnim = ai.TraitInfos<WithTurretAttackAnimationInfo>().Any();
```

Comment:
Shouldn't this be checking the turret the same way as `WithTurretAmmoReloadAnimationInfo`?